SpatialGDK/Source/SpatialGDK/Public/Utils/InterestFactory.h

```diff
@@ -7,53 +7,82 @@
+ * for servers and clients, and if the actor is a player controller, the client worker's interest is also built for that actor.
+ *
+ * The other is server worker interest. Given a load balancing strategy, the factory will take the strategy's defined query constraint
+ * and produce an interest component to exist on the server's worker entity. This interest component contains the primary interest query made
+ * by that server worker.
+ */
+
 class UAbstractLBStrategy;
 class USpatialClassInfoManager;
 class USpatialPackageMapClient;
-class AActor;
 
 DECLARE_LOG_CATEGORY_EXTERN(LogInterestFactory, Log, All);
 
 namespace SpatialGDK
 {
-using ResultType = TArray<Worker_ComponentId>;
+
+struct FrequencyConstraint
+{
+	float Frequency;
+	SpatialGDK::QueryConstraint Constraint;
+};
 
 class SPATIALGDK_API InterestFactory
 {
 public:
-	InterestFactory(AActor* InActor, const FClassInfo& InInfo, const Worker_EntityId InEntityId, USpatialClassInfoManager* InClassInfoManager, USpatialPackageMapClient* InPackageMap);
-
-	static void CreateAndCacheInterestState(USpatialClassInfoManager* ClassInfoManager);
+	InterestFactory(USpatialClassInfoManager* InClassInfoManager, USpatialPackageMapClient* InPackageMap);
 
-	Worker_ComponentData CreateInterestData() const;
-	Worker_ComponentUpdate CreateInterestUpdate() const;
+	Worker_ComponentData CreateInterestData(AActor* InActor, const FClassInfo& InInfo, const Worker_EntityId InEntityId) const;
+	Worker_ComponentUpdate CreateInterestUpdate(AActor* InActor, const FClassInfo& InInfo, const Worker_EntityId InEntityId) const;
 
-	static Interest CreateServerWorkerInterest(const UAbstractLBStrategy* LBStrategy);
+	Interest CreateServerWorkerInterest(const UAbstractLBStrategy* LBStrategy);
 
 private:
+	// Shared constraints and result types are created at initialization and reused throughout the lifetime of the factory.
+	void CreateAndCacheInterestState();
+
 	// Build the checkout radius constraints for client workers
-	static QueryConstraint CreateClientCheckoutRadiusConstraint(USpatialClassInfoManager* ClassInfoManager);
+	// TODO: Pull out into checkout radius constraint utils
+	QueryConstraint CreateClientCheckoutRadiusConstraint(USpatialClassInfoManager* ClassInfoManager);
 	static QueryConstraint CreateLegacyNetCullDistanceConstraint(USpatialClassInfoManager* ClassInfoManager);
 	static QueryConstraint CreateNetCullDistanceConstraint(USpatialClassInfoManager* ClassInfoManager);
-	static QueryConstraint CreateNetCullDistanceConstraintWithFrequency(USpatialClassInfoManager* ClassInfoManager);
+	QueryConstraint CreateNetCullDistanceConstraintWithFrequency(USpatialClassInfoManager* ClassInfoManager);
 
 	// Builds the result types of necessary components for clients
+	// TODO: create and pull out into result types class
 	static ResultType CreateClientNonAuthInterestResultType(USpatialClassInfoManager* ClassInfoManager);
```

Comment:
None of these functions need to be static

Reply:
done